pkgs/top-level/lua-packages.nix

```diff
@@ -71,6 +71,32 @@ let
     };
   };
 
+  luacheck = buildLuaPackage rec {
```

Comment:
For lua packages, your approach is fine. 
We do no have too much lua packages to worry about the size of this file.